src/js/utils/icon.js

```diff
@@ -0,0 +1,13 @@
+import { cloneElement } from 'react';
+
+export const useSizedIcon = (iconArg, size, theme) => {
```

Comment:
Is it too confusing to do:
```
export const useSizedIcon = (icon, size, theme) =>
  (icon && theme?.icon?.matchSize && !icon.props.size && cloneElement(icon, { size })) || undefined;
```

Reply:
I like it -- pushing a change that is along these lines with slight tweak to logic. (need to return icon as is if the conditions return false)